src/Symfony/Component/Yaml/Parser.php

```diff
@@ -730,15 +730,15 @@ private function cleanup($value)
         $this->offset += $count;
 
         // remove leading comments
-        $trimmedValue = preg_replace('#^(\#.*?\n)+#s', '', $value, -1, $count);
+        $trimmedValue = preg_replace('#^\n*\s*(\#.*?\n)+#s', '', $value, -1, $count);
         if ($count == 1) {
             // items have been removed, update the offset
             $this->offset += substr_count($value, "\n") - substr_count($trimmedValue, "\n");
             $value = $trimmedValue;
         }
 
         // remove start of the document marker (---)
-        $trimmedValue = preg_replace('#^\-\-\-.*?\n#s', '', $value, -1, $count);
+        $trimmedValue = preg_replace('#^\n*\s*\-\-\-.*?\n#s', '', $value, -1, $count);
```

Comment:
@stof, moot point. All validators I have found allows these behaviors - they're just trim spaces and empty lines. Also, specification is not saying anything about this. Or I just didn't found.

Let's take a look on empty lines and spaces as on symbols, without a full power to be a symbol with capital letter (like racists :D). They are should not break anything, because not represent anything from itself. IMO.